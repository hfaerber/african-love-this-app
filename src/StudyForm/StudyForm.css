.label-show,
.label-search {
  font-size: .8em;
}

.select-show,
.input-search {
  margin-left: 5px;
  font-size: 16px;
  cursor: pointer;
}

.button-clear-search {
  font-size: 14px;
  cursor: pointer;
  border-radius: 5px;
  padding: 1px 10px;
  cursor: pointer;
}

.div-filter-search-form {
  display: flex;
  align-items: center;
  justify-content: space-around;
  padding: 5px;
  width: 100%
}
