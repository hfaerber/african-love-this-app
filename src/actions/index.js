export const updateUser = user => ({
  type: 'UPDATE_USER',
  user
});

export const updateCountries = countries => ({
  type: 'UPDATE_COUNTRIES',
  countries
});

export const toggleStudyStatus = name => ({
  type: 'TOGGLE_STUDY_STATUS',
  name
});

export const updateSelectedFilter = filter => ({
  type: 'UPDATE_SELECTED_FILTER',
  filter
});

export const updateSearchQuery = query => ({
  type: 'UPDATE_SEARCH_QUERY',
  query
});

export const updateError = errorMessage => ({
  type: 'UPDATE_ERROR',
  errorMessage
});

export const updateLoadingStatus = status => ({
  type: 'UPDATE_LOADING_STATUS',
  status
});
