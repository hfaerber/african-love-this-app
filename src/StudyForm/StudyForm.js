import React, {Component}  from 'react';
import './StudyForm.css';
import { connect } from 'react-redux';
import { updateSelectedFilter } from '../actions';

export class StudyForm extends Component {
  constructor() {
    super();
      this.state = {
        selectedFilter: 'All',
        query: ''
      }
  }

  handleSelection = (event) => {
    this.setState({ selectedFilter: event.target.value });
    this.props.updateSelectedFilter( event.target.value );
  }

  handleSearch = (event) => {
    this.setState({ query: event.target.value });
  }

  handleClick = () => {

  }

  clearSearch = () => {

  }

  render() {
    return(
      <div>
        <label forhtml="display-cards-options" className='label-show'>Show:
          <select name="show" id="display-cards-options" className='select-show'
            value={this.state.selectedFilter}
            onChange={(event) => this.handleSelection(event)}>
            <option value="All">All</option>
            <option value="Still Learning">Still Learning</option>
            <option value="Got It!">Got It!</option>
          </select>
        </label>
        <label forhtml="search-cards" className='label-search'>Search:
          <input type='search' name='query' placeholder='Search Country By Name'
            className='input-search' value={this.state.query}
            onChange={(event) => this.handleSearch(event)}/>
        </label>
        <button className='button-search' onClick={this.handleClick}>
        Search</button>
        <button className='button-clear-search' onClick={this.clearSearch}>
        Clear Search</button>
      </div>
    )
  }
}

// export const mapStateToProps = state => ({
//
// })

export const mapDispatchToProps = dispatch => ({
  updateSelectedFilter: filter => dispatch(updateSelectedFilter(filter))
})

export default connect(null, mapDispatchToProps)(StudyForm)
