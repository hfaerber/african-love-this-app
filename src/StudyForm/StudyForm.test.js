import React from 'react';
import { StudyForm, mapDispatchToProps } from './StudyForm';
import { updateSelectedFilter, updateSearchQuery } from '../actions';
import { shallow } from 'enzyme';

describe('StudyForm', () => {
  describe('StudyForm component', () => {
    let wrapper, mockEvent, mockUpdateSelectedFilter, mockUpdateSearchQuery;

    beforeEach(() => {
      mockUpdateSelectedFilter = jest.fn();
      mockUpdateSearchQuery = jest.fn();
      wrapper = shallow(<StudyForm
        updateSelectedFilter={mockUpdateSelectedFilter}
        updateSearchQuery={mockUpdateSearchQuery}
      />);
      mockEvent = { target: { value: 'testValue' }};
    });

    it('should matchSnapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should invoke handleSelection onChange within select element', () => {
      wrapper.instance().handleSelection = jest.fn();
      wrapper.find('.select-show').simulate('change', mockEvent);
      expect(wrapper.instance().handleSelection).toHaveBeenCalledWith(mockEvent);
    })

    it('should invoke handleSearch onChange within search input', () => {
      wrapper.instance().handleSearch = jest.fn();
      wrapper.find('.input-search').simulate('change', mockEvent);
      expect(wrapper.instance().handleSearch).toHaveBeenCalledWith(mockEvent);
    })

    it('should invoke clearSearch on click of Clear Search button', () => {
      wrapper.instance().clearSearch = jest.fn();
      wrapper.instance().forceUpdate();
      wrapper.find('.button-clear-search').simulate('click');
      expect(wrapper.instance().clearSearch).toHaveBeenCalled();
    })

    it('should update local state when handleSelection is invoked', () => {
      const expected = 'testValue';
      wrapper.instance().handleSelection(mockEvent);
      expect(wrapper.state('selectedFilter')).toEqual(expected);
    })

    it('should invoke updateSelectedFilter prop when handleSelection is invoked', () => {

    })

    it('should update local state when handleSearch is invoked', () => {

    })

    it('should invoke updateSearchQuery prop when handleSearch is invoked', () => {

    })

    it('should update local state when clearSearch is invoked', () => {

    })

    it('should invoke updateSearchQuery when clearSearch is invoked', () => {

    })

  });

  describe('mapDispatchToProps', () => {
    let mockDispatch, mappedProps;
    beforeEach(() => {
      mockDispatch = jest.fn();
      mappedProps = mapDispatchToProps(mockDispatch);
    });

    it('calls dispatch with a updateSelectedFilter action when updateSelectedFilter is called', () => {
      const actionToDispatch = updateSelectedFilter('Got It!');
      mappedProps.updateSelectedFilter('Got It!');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('calls dispatch with a updateSearchQuery action when updateSearchQuery is called', () => {
      const actionToDispatch = updateSearchQuery('Benin');
      mappedProps.updateSearchQuery('Benin');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });
})
