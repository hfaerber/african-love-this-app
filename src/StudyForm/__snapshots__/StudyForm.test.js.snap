// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyForm StudyForm component should matchSnapshot 1`] = `
<div
  className="div-filter-search-form"
>
  <label
    className="label-show"
    forhtml="display-cards-options"
  >
    Show:
    <select
      className="select-show"
      id="display-cards-options"
      name="show"
      onChange={[Function]}
      value="All"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Still Learning"
      >
        Still Learning
      </option>
      <option
        value="Got It!"
      >
        Got It!
      </option>
    </select>
  </label>
  <label
    className="label-search"
    forhtml="search-cards"
  >
    Search All:
    <input
      className="input-search"
      name="query"
      onChange={[Function]}
      placeholder="Search Country By Name"
      type="search"
      value=""
    />
  </label>
  <button
    className="button-clear-search"
    onClick={[Function]}
  >
    Clear Search
  </button>
</div>
`;
